test/new-e2e/system-probe/errors.go

```diff
@@ -104,6 +106,18 @@ var handledErrorsLs = []handledError{
 		metric:      "ssh-connection-refused",
 		action:      retryStack | emitMetric,
 	},
+	{
+		errorType:   ec2InstanceCreateTimeout,
+		errorString: "creating EC2 Instance: operation error",
+		metric:      "ec2-instance-create-timeout",
+		action:      retryStack | emitMetric,
+	},
+	{
+		errorType:   ddAgentRepoFailure,
+		errorString: "Failed to update the sources after adding the Datadog repository.",
+		metric:      "apt-dd-agent-repo-failure",
+		action:      retryStack | emitMetric,
```

Comment:
Do we know if we can gracefully retry from this point? If not, maybe we should just emit a metric right now until retrying is supported.

Reply:
I think we can. I found this error message in [this job](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/636589328#L2500), it was a problem where we ran a job just as the repositories were being updated, and the built-in retry mechanism for [the command](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/636589328#L2308) wasn't enough for it pass. I didn't notice that retries were built-in, but maybe retrying from the stack gives enough time for these rare cases to pass without breaking the test.

Reply:
Will remove the retry in that case until we look further into it